### ISSUE TITLE
Crashed in when formatting a NSError

### ISSUE THREAD

**Issue by xingheng** [CocoaLumberjack/CocoaLumberjack#825]
### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           |  ios
 Platform Version        | 10.2
 CocoaLumberjack Version | e.g. 3.0.0
 Integration Method      | cocoapods
 Xcode Version           | Xcode 8.1
 Repro rate              |  only once
 Demo project link       |  sorry

### Usage
```
NSError *error = ...;

if (error) {
       DDLogError(@"Failed to get data list, error: %@", error);
}
```

### Crash stack:

NSInvalidArgumentException
+[__NSCFConstantString objectForKey:]: unrecognized selector sent to class 0x1b4f13778

CoreFoundation	___exceptionPreprocess + 124
1 libobjc.A.dylib	objc_exception_throw + 56
2 CoreFoundation	___CFExceptionProem
3 CoreFoundation	____forwarding___ + 916
4 CoreFoundation	_CF_forwarding_prep_0 + 92
5 CoreFoundation	__CFErrorCopyUserInfoKey + 184
6 CoreFoundation	__CFErrorFormatDebugDescriptionAux + 52
7 CoreFoundation	_userInfoKeyValueShow + 88
8 CoreFoundation	-[__NSDictionaryI __apply:context:] + 116
9 CoreFoundation	__CFErrorFormatDebugDescriptionAux + 752
10 CoreFoundation	_CFErrorCreateDebugDescription + 56
11 Foundation	-[NSError description] + 12
12 Foundation	_NSDescriptionWithLocaleFunc + 108
13 CoreFoundation	___CFStringAppendFormatCore + 7996
14 CoreFoundation	_CFStringCreateWithFormatAndArgumentsAux2 + 244
15 myapp	+[DDLog log:level:flag:context:file:function:line:tag:format:] (DDLog.m:369)

I checked the issue #735 and output the error message, it seems there is no additional format string in error data string.

Log:
```
Failed to get menu list, error: Error Domain=NSURLErrorDomain Code=-1009 "似乎已断开与互联网的连接。" UserInfo={NSErrorFailingURLStringKey=https://tynb.ourapp.com/api/menu/GetAll?apptoken=puGVOUoocf1PntAGiYbzO/3Dx8utBxABuLfHCYRgc/cP3GHYRsvgOOZD6E4TcIhrT4tTYp5Unf8%3D, _kCFStreamErrorCodeKey=8, NSErrorFailingURLKey=https://tynb.ourapp.com/api/menu/GetAll?apptoken=puGVOUoocf1PntAGiYbzO/3Dx8utBxABuLfHCYRgc/cP3GHYRsvgOOZD6E4TcIhrT4tTYp5Unf8%3D, NSLocalizedDescription=似乎已断开与互联网的连接。, _kCFStreamErrorDomainKey=12, NSUnderlyingError=0x14ea9ed0 {Error Domain=kCFErrorDomainCFNetwork Code=-1009 "(null)" UserInfo={_kCFStreamErrorDomainKey=12, _kCFStreamErrorCodeKey=8}}}
```

Besides, I tried to repo it when debugging but failed, it only crashed once in our release build currently.

**Comment by bpoplauschi** [comment]
This is the same issue as in #735 - closing in favour of that one. And try the solutions described there. Your problem is the string contains some special strings used for formatting.
